src/esmockModule.js

```diff
@@ -92,14 +98,24 @@ const esmockModuleCreate = async (treeid, def, id, fileURL, opt) => {
   return mockModuleKey
 }
 
+const esmockMetaResolve = async (id, parent) => {
+  try {
+    return decodeURI(await import.meta.resolve(id, asFileURL(parent)))
+  } catch (e) {
+    return null
+  }
+}
+
 const esmockModuleId = async (parent, treeid, defs, ids, opt, mocks, id) => {
   ids = ids || Object.keys(defs)
   id = ids[0]
   mocks = mocks || []
 
   if (!id) return mocks
 
-  const fileURL = resolvewith(id, parent)
+  const fileURL = isMetaResolve
+    ? await esmockMetaResolve(id, parent)
+    : resolvewith(id, parent)
```

Comment:
`resolvewithplus` has to be always installed - even when using the `import.meta.reslove`?

IMHO, we can change it to an optional dependency later - so Node.js v18 users won't have to install it.

Reply:
Thanks for the review. Per https://github.com/iambumblehead/esmock/issues/176#issuecomment-1283357993 I'm planning a follow-up PR that makes resolvewithplus an optional dependency and, if tests pass and there are no objections, merge and publish.

Reply:
> we can change it to an optional dependency later 

Do you think it is OK to try making it an optional dependency sooner rather than later? Do you foresee any problem? I have not used optional dependencies before.

Reply:
it's just to make esmock a smaller install size, but given `resolvewithplus` is pretty small, and I suppose most users are still using node.js < 18, it seems not a hurry thing. :)

Reply:
The reason for sooner: yarn-PnP requires zero-dependency loader packages. [This issue reports][0] esmock requires a yarn-PnP 'bundling' step, but if resolvewithplus becomes an optional dependency, esmock will be yarn-PnP-usable out-of-box no configuration necessary

[0]: https://github.com/iambumblehead/esmock/issues/176

Reply:
well. as I know, npm will install optional deps by default, unless `--no-optional` was passed.

I'm not very familiar with yarn pnp, will it bundle esmock and its deps?  My guess is no, so it won't work even making the `resolvewithplus` to be optional.

 To make esmock a 0-dependency, we can:
1. use something like `rollup` to bundle;
2. embed resolvewithplus inside esmock (as metioned in https://github.com/iambumblehead/esmock/issues/176#issuecomment-1281549228)

Reply:
> IMHO, we can change it to an optional dependency later - so Node.js v18+ users won't have to install it.
Agree to do it a bit later - optional deps are gray zone, due to very different scenarios * package managers specific behaviour * existing bugs / issues.

Reply:
This was useful discussion (thank you). Embedding resolvewithplus.js inside the esmock package seems the easiest way to get a zero-dependency esmock package now

Reply:
resolvewithplus can be embedded with simple sed and cp commands because it is a single file